source/grid/grid_generator.cc

```diff
@@ -8556,6 +8572,14 @@ namespace GridGenerator
             for (const auto &cell_vertices : table_2D_cell)
               add_cell(dim, cell_vertices, material_id_cell, manifold_id_cell);
           }
+        else if (dim == 3)
+          {
+            // get cell-manifold id from current quad cell
```

Comment:
Can you take this line and move it up outside the if? Nor reason to duplicate it.